main/src/ca/uwaterloo/flix/language/phase/ClosureConv.scala

```diff
@@ -279,7 +266,8 @@ object ClosureConv {
     case Expression.NewObject(clazz, tpe, purity, methods0, loc) =>
       val methods = methods0 map {
         case JvmMethod(ident, fparams, exp, retTpe, purity, loc) =>
-          JvmMethod(ident, fparams, visitExp(exp), retTpe, purity, loc)
+          val closure = createLambdaClosure(fparams, exp, retTpe, loc)
+          JvmMethod(ident, fparams, closure, retTpe, purity, loc)
```

Comment:
I think the type used here for `JvmMethod` might be wrong. See:

If we have `x -> x + 1`, a lambda, then it has type `Int32 -> Int32`.

If we write:

```
object Foo {
  def inc(x: Int32): Int32 = x + 1 
}
```
then the type of `exp` will be Int32, not `Int32 -> Int32.

Reply:
Thus here we must build a function type from the type of the fparams and retType.

Reply:
I think that this comment is now moot given that the closure type is now gone?

Reply:
I think so.